sys/dev/drm/panfrost/panfrost_drv.c

```diff
@@ -728,9 +728,23 @@ panfrost_irq_hook(void *arg)
 
 	sc->drm_dev.dev_private = sc;
 
-	panfrost_device_init(sc);
-	panfrost_mmu_init(sc);
-	panfrost_job_init(sc);
+	err = panfrost_device_init(sc);
```

Comment:
Does this hook get retried if it returns?  Otherwise, I'm not sure how adding these checks is making sure the driver is waiting for the kernel scheduler (which should always be reading if an irq_hook is running I imagine?) before it calls this function?

Reply:
panfrost_job_init is actually tries to start scheduler

Reply:
sorry, I am talking about DRM scheduler!

Reply:
Perhaps update the commit log to say "DRM scheduler" explicitly then?

Reply:
(Though I think a more accurate log might be something like "Don't ignore errors when initializing panfrost")

Reply:
Fixed. Thanks!